lua/entities/gmod_wire_expression2/core/selfaware.lua

```diff
@@ -106,12 +106,35 @@ e2function void setName( string name )
 	e:SetOverlayText(name)
 end
 
--- Get the name of another E2
+-- Set the name of a entity (component name if not E2). Thanks CaptainPRICE for idea and basic implementation
+e2function void entity:setName( string name )
+	if not IsValid(this) or E2Lib.getOwner(self, this) ~= self.player then return end
+	if this:GetClass() == "gmod_wire_expression2" then
+		if this.name == name then return end
+		if name == "generic" or name == "" then
+			name = "generic"
+			this.WireDebugName = "Expression 2"
+		else
+			this.WireDebugName = "E2 - " .. name
+		end
+		this.name = name
+		this:SetNWString( "name", this.name )
+		this:SetOverlayText(name)
+	else
+		if ( this.wireName and this.wireName == name ) or string.find(name, "[\n\r\"]") ~= nil then return end
```

Comment:
Now if I rethink your "fix", what's actually the point of checking `this.wireName`? If it is nil (like it is initially), then it won't be equal to `name`, and `this.wireName` would be set afterwards. Then, the second time it called, it won't be nil value (then it actually compares the two). Basically meaning you just **added a redundant check** (made the code worse). **Removed a localized function** `string.find` go (again made the code worse)?!
**No actual bug fixes from what I see here**.
I would call this `simpler, but also worse implementation`. What are you even doing?!